backoff/_wait_gen.py

```diff
@@ -47,3 +47,15 @@ def constant(interval=1):
     """
     while True:
         yield interval
+
+
+def predefined(intervals=[1]):
+    """Generator for predefined intervals.
+
+    Args:
+        intervals: A list with the intervals to yield.
+    """
+    while True:
+        if len(intervals):
```

Comment:
I'd rather have the intervals be defined by an arbitrary iterable rather than requiring a mutable list.  All the other places in the API which take a sequence are type-agnostic.

This way any iterable would work including generators.

   @backoff.on_exception(backoff.predefined, intervals=(1, 2, 3))

And it would play nice with `itertools`. The equivalent of backoff.constant with interval=1:

   @backoff.on_exception(backoff.predefined, intervals=itertools.repeat(1))

And I don't know why you'd want to do this, but you could:

    @backoff.on_exception(backoff.predefined, intervals=itertools.cycle([1,2,3]))

We might still want to support the behavior you defined where the last item is repeated once the iterator is exhausted, so something like:

    itr = iter(intervals)
    while True:
        try:
            last = next(itr)
        except StopIteration:
            pass
        yield last

But this has me thinking a little bit... what if instead, a `StopIteration` raised by the wait generator was actually a signal to trigger giving up?

    @backoff.on_exception(backoff.predefined, intervals=[1,3,6])

In other words, the above definition would result in a maximum of 4 tries with those predefined intervals between them. I feel like this might be a convenient way to fully specify retry behavior with a built in give up condition.